src/main/java/org/apache/maven/plugins/javadoc/JavadocUtil.java

```diff
@@ -1815,7 +1831,7 @@ private static CloseableHttpClient createHttpClient( Settings settings, URL url
         builder.setUserAgent( "Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.0)" );
 
         // Some server reject requests that do not have an Accept header
-        builder.setDefaultHeaders( Arrays.asList( new BasicHeader( HttpHeaders.ACCEPT, "*/*" ) ) );
+        builder.setDefaultHeaders( Collections.singletonList( new BasicHeader( HttpHeaders.ACCEPT, "*/*" ) ) );

```

Comment:
stet. Immutability makes Collections.singletonList not semantically equivalent to Arrays.asList

Reply:
> Immutability makes Collections.singletonList not semantically equivalent to Arrays.asList
Yes but I don't think a immutable collection used here can be any problem.
I see no change to this collection after passed in as parameter here(in the codes).
Or please give me an example where can it break.

Reply:
The API accepts a list, not an immutable list, and does not guarantee that it will not try to mutate that list.

Reply:
@elharo fine, will revert this.